### DISCUSSION TITLE
How to Add Security/Authentication for Prefect Core Server Deployed on GCP?

### DISCUSSION THREAD

**Discussion by rileyhun** [PrefectHQ/prefect#3068] [Q&A] [answered]
Hi everyone,

I'm new to Prefect, and have been following an article [1] that described how to set up the UI on GCP. I've managed to set it up successfully on Compute Engine. The next step I want to integrate is some sort of authentication. Even basic authentication will do. I understand Prefect Cloud has authentication built in, but Prefect Core doesn't have anything of the sort. Any ideas on how to facilitate this? 

Many thanks in advance!

[1] https://medium.com/the-prefect-blog/prefect-server-101-deploying-to-google-cloud-platform-47354b16afe2

**Comment by cicdw** [comment]
Hi @rileyhun - your best bet would be to put up some sort of network security layer in front of the Apollo endpoint and the UI endpoint or host it within a private network.

**Comment by richard-barella** [comment, marked as answer]
I think this page has a possible solution https://docs.prefect.io/v3/advanced/security-settings

**Reply by zzstoatzz** [reply]
yes OSS server didn't have auth before, but now it does natively
